tools/generate-mocks/generate-mocks.ts

```diff
@@ -385,25 +435,39 @@ export class ${mockClassName} {${propertiesString}${methodsString}}
     return components;
   }
 
-  private visitTree(node: ts.Node, componentMetaData?: ComponentMetaData) {
+  private visitTree(
+    node: ts.Node,
+    componentMetaData?: ComponentMetaData,
+    assertedTypesMap?: Map<string, string>
```

Comment:
The thing is we only need the assertedTypesMap (aliasMap) when we encounter a class decorator, where we want to swap out the classname in componentMetaData when it is in the aliasMap.

At all other times, we rely on the aliasesMap not being present, because we then might be on an angular input decorator or similar, and then we dont want to extract the metadata from the custom-elements.json. 
Which is why the recursive visitTree call does not have the aliasesMap.

It is a bit convoluted and I will try to think up a way to simplify it.